app/modules/set-selector/genre-detail.controller.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+require('./set-selector.module.js')
+    .controller('GenreDetailController', GenreDetailController);
+
+// @ngInject
+function GenreDetailController($scope, $state, $stateParams, GenresFactory) {
+    var vm = this;
+    vm.go = go;
+    vm.loading = true;
+
+    GenresFactory.detail($stateParams.genreId)
+        .then(function (genre) {
+            vm.genre = genre;
+            vm.moreGenres = GenresFactory.list(3);
+            vm.loading = false;
+            // TODO: fix this ugly business
+            if (!$scope.$$phase) {
+                $scope.$digest();
+            }
```

Comment:
I've found a way to do this. Remove the `if` block, and wrap the `vm` stuff in a timeout:
```
GenresFactory.detail($stateParams.genreId)
    .then(function (genre) {
        $timeout(function () {
            vm.genre = genre;
            vm.moreGenres = GenresFactory.list(3);
            vm.loading = false;
        });
    });
```

should do it..?